From 247b5a9555a86de532de68d26b9c3b52fe5a1cf8 Mon Sep 17 00:00:00 2001
From: Jens Nyberg <jens.nyberg@gmail.com>
Date: Fri, 6 Jan 2023 16:01:54 +0100
Subject: [PATCH] Fix counter and freq for libretro

This is the follow up to my previous fix to make sure libretro
works as well.
---
 src/system/libretro/tic80_libretro.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/src/system/libretro/tic80_libretro.c b/src/system/libretro/tic80_libretro.c
index 923d3de05..95e0b1df6 100644
--- a/src/system/libretro/tic80_libretro.c
+++ b/src/system/libretro/tic80_libretro.c
@@ -77,6 +77,22 @@ struct tic80_state
 };
 static struct tic80_state* state;
 
+/**
+ * TIC-80 callback; Request counter
+ */
+static u64 tic80_libretro_counter()
+{
+    return clock();
+}
+
+/**
+ * TIC-80 callback; Request freq
+ */
+static u64 tic80_libretro_freq()
+{
+    return CLOCKS_PER_SEC;
+}
+
 /**
  * TIC-80 callback; Requests the content to exit.
  */
@@ -806,7 +822,7 @@ void tic80_libretro_update(tic80* game)
 	tic80_libretro_update_keyboard(&state->input.keyboard);
 
 	// Update the game state.
-	tic80_tick(game, state->input, NULL, NULL);
+	tic80_tick(game, state->input, tic80_libretro_counter, tic80_libretro_freq);
 	tic80_sound(game);
 }
 
